incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/openapi/http/proxy/OpenApiHttpProxyConfigGenerator.java

```diff
@@ -0,0 +1,505 @@
+                    .build()
+                .inject(this::injectPlainTcpRoute)
+                .inject(this::injectTlsTcpRoute)
+                .inject(this::injectPlainAndTlsTcpRoutes)
+                .build()
+            .inject(this::injectTlsServer)
+            .binding()
+                .name("http_server0")
+                .type("http")
+                .kind(SERVER)
+                .options(HttpOptionsConfig::builder)
+                    .access()
+                        .policy(CROSS_ORIGIN)
+                        .build()
+                    .inject(this::injectHttpServerOptions)
+                    .build()
+                .inject(this::injectHttpServerRoutes)
+                .build()
+            .binding()
+                .name("http_client0")
+                .type("http")
+                .kind(CLIENT)
+                .inject(this::injectHttpClientRoute)
+                .build()
+            .inject(this::injectTlsClient)
+            .binding()
+                .name("tcp_client0")
+                .type("tcp")
+                .kind(CLIENT)
+                .options(TcpOptionsConfig::builder)
+                    .host("") // env
+                    .ports(new int[]{0}) // env
+                    .build()
+                .build()
+            .inject(this::injectGuard)
+            .inject(this::injectVaults)
+            .build();
+    }
+
+    private BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>> injectPlainTcpRoute(
+        BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>> binding)
+    {
+        if (isPlainEnabled && !isTlsEnabled)
+        {
+            binding
+                .route()
+                    .exit("http_server0")
+                    .build();
+        }
+        return binding;
+    }
+
+    private BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>> injectTlsTcpRoute(
+        BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>> binding)
+    {
+        if (isTlsEnabled && !isPlainEnabled)
+        {
+            binding
+                .route()
+                    .exit("tls_server0")
+                    .build();
+        }
+        return binding;
+    }
+
+    private BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>> injectPlainAndTlsTcpRoutes(
+        BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>> binding)
+    {
+        if (isPlainEnabled && isTlsEnabled)
+        {
+            binding
+                .route()
+                    .when(TcpConditionConfig::builder)
+                        .ports(httpPorts)
+                        .build()
+                    .exit("http_server0")
+                    .build()
+                .route()
+                    .when(TcpConditionConfig::builder)
+                        .ports(httpsPorts)
+                        .build()
+                    .exit("tls_server0")
+                    .build();
+        }
+        return binding;
+    }
```

Comment:
Suggest we keep it simple, and always include the port-specific route for `http` or`https`.
So no need for 3 cases of `tls && !plain`, `!tls && plain` and `tls && plain`, just 2 cases `(optional) plain` and `(optional) tls`.

Reply:
done